diesel_derives2/src/from_sql_row.rs

```diff
@@ -0,0 +1,57 @@
+use quote::Tokens;
+use syn;
+
+use meta::*;
+use util::*;
+
+pub fn derive(mut item: syn::DeriveInput) -> Result<Tokens, Diagnostic> {
+    let flags =
+        MetaItem::with_name(&item.attrs, "diesel").unwrap_or_else(|| MetaItem::empty("diesel"));
+    let struct_ty = ty_for_foreign_derive(&item, &flags)?;
+
+    item.generics.params.push(parse_quote!(__ST));
+    item.generics.params.push(parse_quote!(__DB));
+    {
```

Comment:
Why is this in a block ?

Reply:
`where_clause` is an exclusive reference to a member of `item.generics`. The line after this block ends creates a shared reference to `item.generics`. I believe this block can go away when NLL lands.